### DISCUSSION TITLE
Nilsson Ring

### DISCUSSION THREAD

**Discussion by mahrud** [Macaulay2/M2#1070] [Q&A]
As a part of the D-modules coding sprint, we want to compute solutions that live in the Nilsson ring:
![image](https://user-images.githubusercontent.com/3147718/73404988-7d30fe80-42b8-11ea-9214-50fa8fede635.png)
(Page 95 of Grobner Deformations of Hypergeometric Differential Equations, Saito Sturmfels, Takayama)

Here e^i are rational exponent vectors, not powers of e. Essentially, we want sums of polynomials in x,y,z,... (with rational exponents) and logarithms log(x),log(y),log(z),...

Currently, we're thinking of two new types: `NilssonTerm` and `NilssonSeries`:
```
NilssonTerm = new Type of HashTable
t = new NillsonTerm from {
  MonomialExponents => {1/2,0},
  PolynomialInLog   => xy
}
```
Here MonomialExponents is the exponent vector of the x^(1/2)y^0 and PolynomialInLog represents log(x)log(y).

**Question 1**: is there a smart way to present this? (ie. net NilssonTerm)
The main problems are logs and rational exponents.

We also need to add and multiply these terms to get truncated series:
```
NilssonSeries = new Type of List
new NillsonSeries from {
  NilssonTerm {..},
  NilssonTerm {..},
  ...
}
```

**Question 2**: is there a way to define a monoid with objects of a given type, in this case NilssonTerms?

**Comment by christineBerkesch** [comment]
Once this is done, see Theorem 2.3.11 in SST for a description of the log part of an indicial solution
